zebra/zebra_wrap_script.c

```diff
@@ -0,0 +1,290 @@
+/*
+ * Zebra Script Wrapper
+ * Copyright (C) 2018  6WIND
+ *
+ * FRR is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2, or (at your option) any
+ * later version.
+ *
+ * FRR is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with FRR; see the file COPYING.  If not, write to the Free
+ * Software Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
+ * 02111-1307, USA.
+ */
+
+#include <zebra.h>
+
+#include "json.h"
+
+#include "zebra/debug.h"
+#include "zebra/zebra_wrap_script.h"
+
+/* this struct temporarily stores the list of headers
+ * - name is used to store the name of the header field
+ * - attribute can be used to store values that have no column
+ * assigned ( extra param)
+ * ex:
+ *  columnA   columnB       columnC    columnD
+ *    <val1>   <val2>        <val3>     <val4>       <val5> <val6>..
+ *    <val7>   <val8>        <val9>     <val10>    <val11> <val12>..
+ * columnA .. columnD are stored in name field
+ * attribute us used to store <val5>..<val6> or <val11>..<val12>
+ */
+struct item_list {
+	char *name;
+	char *attribute;
+};
+
+#define DATA_LEN 4096
+#define ITEM_MAXIMUM 15
+#define DATA_LINE_MAX 200
+
+/* debug information
+ */
+#define SCRIPT_DEBUG	(1<<1)
+#define SCRIPT_ITEM_LIST	(1<<2)
+#define SCRIPT_ELEMENT_LIST	(1<<3)
+
+static int zebra_wrap_debug;
+
+
+void zebra_wrap_init(void)
+{
+	zebra_wrap_debug = 0;
+}
+
+static int search_current_word(char *current_str, int init,
```

Comment:
not yet. I think strsep or strchr could be helpful.

for iptable output process, I think of keeping the name very generic, so that this kind of function can be applied to not only iptables but also other functions ( ipset ?)